### DISCUSSION TITLE
Hardware acceleration on GPUs with decoding only (e.g. RX 6400)

### DISCUSSION THREAD

**Discussion by RedAnon** [blakeblackshear/frigate#4753] [Ask A Question] [answered]
Hi,
I'm about to deploy Frigate (with M.2 coral board) on a spare Xeon board I have. Since this particular Xeon doesn't have an iGPU I probably also need a GPU that can decode x265 (as that's what's coming out of the cameras) streams. **Or do I?**

Question is will the entry point GPUs that lack video encoding and only have video decoding such as the AMD Radeon RX 6400 do?
What's the rule of thumb on GPUs? How much vram does one need per camera per 4k stream?

**Comment by NickM-27** [comment, marked as answer]
You absolutely want hardware to decode the streams. Encoding isn't done unless you're looking to transcode from h265 to h264 

Also, decoding is only done on the detect stream which we highly recommend to be a sub stream so you should definitely not be decoding 4K

**Reply by RedAnon** [reply]
> You absolutely want hardware to decode the streams.

ok but how many streams can a GPU handle?

> Also, decoding is only done on the detect stream which we highly recommend to be a sub stream so you should definitely not be decoding 4K

If it's low res then it surely can't be THAT hardware intensive can it?

**Reply by NickM-27** [reply]
> ok but how many streams can a GPU handle?

Completely depends on the GPU, can range anywhere from 3-4 to 20 stream decoding sessions, that is something that is going to be a spec of the GPU itself.

Most GPUs can only handle 2 or 3 encoding sessions (but frigate doesn't use encode for anything right now, unless you want to transcode your h265 camera to h264 which isn't really needed)

> If it's low res then it surely can't be THAT hardware intensive can it?

well if you're talking many cameras then it adds up. Also the problem is not ONLY hardware intensity, also hardware latency. If your CPU is slower to decode the stream than a GPU, that extra time means more latency between what is happening and when frigate sees the object come into the frame.

For example, my CPU is a Ryzen 5700G which has an integrated GPU and when I use that to decode a stream in frigate there is a ~2 second delay between the real world and what frigate sees. I also have an nvidia 3050 and when I use that to decode a stream in frigate there is a ~ 750 ms delay between the real world and what frigate sees. I use both, nvidia for high priority cameras and amd iGPU for low priority cameras. 

I don't think there's a harm in just trying your CPU first and seeing how it works, then make the decision based on that.
